src/cluster/replication.cc

```diff
@@ -601,14 +658,15 @@ ReplicationThread::CBState ReplicationThread::incrementBatchLoopCB(bufferevent *
         if (!s.IsOK()) {
           error("[replication] CRITICAL - Failed to write batch to local, {}. batch: 0x{}", s.Msg(),
                 util::StringToHex(batch.Data()));
-          return CBState::RESTART;
+          return {CBState::RESTART, data_written};
```

Comment:
I think we don't need to add an extra method and return value, just add a function like `sendReplConfAck` after applying the write batch.

Reply:
get it, I updated it here: https://github.com/apache/kvrocks/pull/3061/commits/9d44339c4d130923f84716bf3bddb73e6ba3846d.
I don't think we should send ack after applying each write batch though, because if there are several batches in one read callback, it is more efficient to write all of the batches and ack once.